tracing/zipkin/zipkin.go

```diff
@@ -259,6 +312,12 @@ func FromContext(ctx context.Context) (*Span, bool) {
 	return span, true
 }
 
+// CollectorToContext stores a Zipkin collector to be used when annotating
+// resources using their own child spans with NewChildSpan.
+func CollectorToContext(ctx context.Context, c Collector) context.Context {
+	return context.WithValue(ctx, CollectorContextKey, c)
+}
```

Comment:
The Zipkin Collector needs to be available inside service methods when creating a child span with NewChildSpan(). Best transport is context. By wiring your Zipkin collector in main to the root context using this function, you can now use NewChildSpan() inside service methods having a copy of context.

Example of NewChildSpan:

```go
// Foo example method of our Demo Service
func (demo *Demo) Foo(ctx context.Context, a, b string) (string, error) {
	// First we'll need to retrieve the span.
	span, collect := zipkin.NewChildSpan(
		ctx,
		"query", 
		zipkin.ServerAddr("my.mysql.host:3306", "MySQL"),
	)
	// Now we can annotate this resource
	span.Annotate("query start")
	span.AnnotateBinary("query", "select * from something where someotherthing = 'somevalue' order by bla limit 100")
	// run the query...

	span.Annotate("query end")
	span.Annotate("query start")
	/// do more and annotate...

	// we are done with this span... time to collect
	collect()
}
```